src/serverlessApigatewayLogRetentionPlugin.js

```diff
@@ -9,6 +10,9 @@ class ApigatewayLogRetentionPlugin {
         this.hooks = {
             'after:deploy:deploy': this.setApigatewayLogRetention.bind(this, serverless),
         };
+        AWS.config.update({
```

Comment:
not the best place to do this as it would always set the proxy even if http_proxy is not exported? Needs to be taken out into a separate method which checks if "HTTP_PROXY" is set and only then update the config

Reply:
also this is the suggested approach by AWS https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/node-configuring-proxies.html